mlrun/launcher/factory.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2023 MLRun Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import mlrun.config
+import mlrun.errors
+import mlrun.launcher.base
+import mlrun.launcher.local
+import mlrun.launcher.remote
+
+
+class LauncherFactory(object):
+    @staticmethod
+    def create_launcher(
```

Comment:
Is there any advantage to creating a new launcher every time this is called? Aren't the launchers basically singletons in nature? Meaning, there shouldn't be more than 1 server-side launcher, 1 client-local launcher etc.?

Reply:
I will add a follow-up PR to change the classes to static classes / singletons